controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -708,7 +710,11 @@ void wfa_ca::handle_wfa_ca_message(
                 reply(sd, cmdu_tx, eWfaCaStatus::INVALID, err_string);
                 return;
             }
-
+            //remove all old configs for the AL-MAC address when we get a DEV_SET_CONFIG command
+            if (!old_configs_removed) {
+                database.clear_bss_info_configuration(net::network_utils::mac_from_string(al_mac));
+                old_configs_removed = true;
+            }
```

Comment:
What is the purpose of the flag `old_configs_removed` and the `if` statement? 
`old_configs_removed` is always set to `false` a couple of lines before, so the code inside the if statement will always happen, so it makes the `if` statement redundant. Also in the `if` scope you change the value old_configs_removed  to `true` but it is not used afterward so it turns out that this flag is redundant too.

Reply:
Oh sorry, you are right. I didn't notice that the `old_configs_removed` is defined outside the loop.
But still, this solution has a hole in it.
Since you used a flag, which you set it one time, it will create a problem if there are more than one AL_MACs. So, in that case, it will work for the first AL_MAC, but for the second, third and go on, it will not delete the configuration.
I suggest creating unordered_set of al_mac which we have cleared configuration on:

```cpp
std::unordered_set<std::string> al_mac_cleared_conf;
// code 
// for loop
    // code
    if (al_mac_cleared_conf.find(al_mac) == al_mac_cleared_conf.end()) {
          database.clear_bss_info_configuration(net::network_utils::mac_from_string(al_mac));
          al_mac_cleared_conf.insert(al_mac);
    }
```